ui/src/containers/BeamlineSetupContainer.jsx

```diff
@@ -261,11 +280,11 @@ function BeamlineSetupContainer(props) {
             <InOutSwitch
               openText="Power On"
               offText="Power Off"
-              openValue="PowerOn"
-              offValue="PowerOff"
+              openValue="READY"
+              offValue="DISABLED"
               labelText="Sample Changer"
               value={sampleChanger.state}
-              onSave={sendCommand}
+              onSave={handleSendCommand}
             />
```

Comment:
I would suggest :

`            <InOutSwitch
              openText="Power On"
              offText="Power Off"
              openValue="PowerOff"
              offValue="PowerOn"
              labelText="Sample Changer"
              value={sampleChanger.state}
              onSave={sendCommand}            
/>
`
As you powerOn when state is READY
just  switch openValue to offValue would solve this,  
Maybe  ??

but I definitely would prefer to not have a another handle method handleSendCommand
InOutSwitch already has a Handle method in.

Reply:
I don't understand how that will work.

Sample Changers don't have PowerOff nor PowerOn states. The go between READ and DISABLED.

Reply:
> I would suggest :
> 
> ` <InOutSwitch openText="Power On" offText="Power Off" openValue="PowerOff" offValue="PowerOn" labelText="Sample Changer" value={sampleChanger.state} onSave={sendCommand} />` As you powerOn when state is READY just switch openValue to offValue would solve this, Maybe ??
> 
> but I definitely would prefer to not have a another handle method handleSendCommand InOutSwitch already has a Handle method in.

Consider this as Not relevant:  see my other response on top,